packages/compass-global-writes/src/components/shard-zones-table.tsx

```diff
@@ -8,47 +8,189 @@ import {
   Row,
   Cell,
   css,
+  useLeafyGreenTable,
+  type LGColumnDef,
+  type LeafyGreenTableRow,
+  flexRender,
+  type HeaderGroup,
+  SearchInput,
+  type LGTableDataType,
+  getFilteredRowModel,
+  type LgTableRowType,
 } from '@mongodb-js/compass-components';
 import type { ShardZoneData } from '../store/reducer';
 
 const containerStyles = css({
-  maxWidth: '700px',
   height: '400px',
 });
 
+interface ShardZoneRow {
+  locationName: string;
+  zone: string;
+}
+
+interface ShardZoneExpandableRow extends ShardZoneRow {
+  subRows: ShardZoneRow[];
+}
+
+const columns: Array<LGColumnDef<ShardZoneRow>> = [
+  {
+    accessorKey: 'locationName',
+    header: 'Location Name',
+    enableSorting: true,
+  },
+  {
+    accessorKey: 'zone',
+    header: 'Zone',
+    enableSorting: true,
+    enableGlobalFilter: false,
+  },
+];
+
+const parseRow = ({
+  isoCode,
+  readableName,
+  zoneName,
+  zoneLocations,
+}: ShardZoneData): ShardZoneRow => ({
+  locationName: `${readableName} (${isoCode})`,
+  zone: `${zoneName} (${zoneLocations.join(', ')})`,
+});
+
+const parseData = (shardZones: ShardZoneData[]): ShardZoneExpandableRow[] => {
+  const grouppedZones = shardZones.reduce<
+    Record<ShardZoneData['typeOneIsoCode'], ShardZoneExpandableRow>
+  >((groups, next) => {
+    const { typeOneIsoCode, isoCode } = next;
+    if (!groups[typeOneIsoCode]) {
+      groups[typeOneIsoCode] = { ...parseRow(next), subRows: [] };
+    }
+    if (typeOneIsoCode === isoCode) {
+      Object.assign(groups[typeOneIsoCode], parseRow(next));
+    } else {
+      groups[typeOneIsoCode].subRows.push(parseRow(next));
+    }
+    return groups;
+  }, {});
+  return Object.values(grouppedZones);
+};
+
+const hasFilteredChildren = (
+  row: LgTableRowType<LGTableDataType<ShardZoneRow>>
+) =>
+  row.subRows.filter(
+    (subRow) => Object.values(subRow.columnFilters).includes(true) // columnFilters: e.g. { __global__: true }
+  ).length > 0;
```

Comment:
Small nit, we have a builtin array method for this specifcic use case:

```suggestion
  row.subRows.some(
    (subRow) => Object.values(subRow.columnFilters).includes(true) // columnFilters: e.g. { __global__: true }
  );
```

Reply:
Hah, just saw Sergey already mentioned this :)